test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -108,17 +108,16 @@ TEST_F(FileSystemImplTest, FileReadToEndDoesNotExist) {
                EnvoyException);
 }
 
-#ifndef WIN32
-// In Windows this method of removing the permissions does not make read fail.
-TEST_F(FileSystemImplTest, FileReadToEndNotReadable) {
-  const std::string data = "test string\ntest";
-  const std::string file_path = TestEnvironment::writeStringToFileForTest("test_envoy", data);
-
-  std::filesystem::permissions(file_path, std::filesystem::perms::owner_all,
-                               std::filesystem::perm_options::remove);
-  EXPECT_THROW(file_system_.fileReadToEnd(file_path), EnvoyException);
-}
-#endif
+// Issue https://github.com/envoyproxy/envoy/issues/25614
```

Comment:
Instead of commenting out the whole test (which changes all the lines in the test) how about changing `#ifndef WIN32` to `#if 0` or some such?

Reply:
@RyanTheOptimist : thats the proposed approach by @ravenblackx and @phlax . I would have kept it to previous state with issue link. Just that different suggestions. Can you guys consent on whats needed. I dont think this issue needs to spent too much on the brain power. As long as issue ref is linked we are good. Please let me know before I push any other changes.

Reply:
I don't mind which way the test is disabled, so long as it's linked to the issue and the code in the test is the version that had coverage but didn't work in some environments, not the version that is a duplicate of an existing test with a name that says it's testing something it's not. :)

Another way to disable it would be to add one of the skip-the-test-and-log-it macros, which might be even better. But I am happy with this as-is.

Reply:
It's `GTEST_SKIP()` in case you want to do it that way. That would also play well with any future attempt to make it conditional, since then it could be simply changed to.
```
if (however we detect the test is not going to work) {
  GTEST_SKIP();
}
```

Reply:
If we want to disable the test so it doesn't run, then we should just add `DISABLED_` to the name. So that would be:

```
TEST_F(FileSystemImplTest, DISABLED_FileReadToEndNotReadable) {
```

However, I'm still unclear if we've tried to make the test conditional or not. If we make it conditional, we should use `GTEST_SKIP()` as @ravenblackx suggests. If we can make this test conditional that would be much better. @noah8713 have you tried the suggestion here:

> It's possible the test could be conditional (check that the permissions preclude reading the file after they've been changed, and if they don't, just log that the test was skipped and don't do the EXPECT), which would provide the coverage in the coverage environment where the test works, and then 'skip' in the environment where the test doesn't work.

Reply:
Ping

Reply:
I think the idea here is making it *not broken* is important enough that landing this without quibbling about the best way to make the test conditional would be worthwhile, with fixing it better being a thing to do later.

A big part of the problem with making it conditional is that we're not even sure what would work, at this point, and it's hard to test because in most environments the test works already - getting it to run in the environment where it doesn't work is tricky, which makes iterating towards a better solution tricky.

Reply:
If we're going to land without making test conditional, then lets use `DISABLED_` to disable the test, as that's the supported pattern for doing so.

Reply:
Thanks @ravenblackx and @RyanTheOptimist : made it DISABLED_ and rebased. PTAL

Reply:
Thanks. In the future, please do not use rebase and force push, per the guidance in CONTRIBUTING.md:

https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md

> Once your PR is under review, please do not rebase it. If you rebase, you will need to force push to github, and github's user interface will force your reviewer to review the PR from scratch rather than simply look at your latest changes. It's much easier to review new commits and/or merges. We squash rebase the final merged commit so the number of commits you have in the PR don't matter. Again once your PR is assigned a reviewer, unless you need to fix DCO please do not force push. If you need to pull recent changes you can run
>
> ```
> branch=$(git status|head -1|cut -f3 -d\ )
> git checkout main
> git pull
> git checkout "$branch"
> git pull
> ```

Reply:
Ack! Thanks a bunch. Noted!